app/controllers/carts_controller.rb

```diff
@@ -1,15 +1,21 @@
 class CartsController < ApplicationController
+  before_action :check_logged_in_user, only: :add_to_cart
 
   def index
     @cart = get_cart
   end
 
   def add_to_cart
-    if logged_in?
-      add_product_to_cart_cookie(params[:product_id])
-      redirect_to root_path
-    else
-      redirect_to login_path
-    end
+    add_product_to_cart(params[:product_id])
+    redirect_to root_path
```

Comment:
Here you call the method add_product_to_cart with parenthesis and redirect_to without them

Reply:
Thanks!